src/Cmd/Add.cc

```diff
@@ -0,0 +1,235 @@
+#include "Add.hpp"
+
+#include "../Logger.hpp"
+#include "../Manifest.hpp"
+#include "Common.hpp"
+
+#include <algorithm>
+#include <cstdlib>
+#include <string>
+#include <toml.hpp>
+#include <vector>
+
+static int addMain(std::span<const StringRef> args);
+
+const Subcmd ADD_CMD =
+    Subcmd{ "add" }
+        .setDesc("Add dependencies to poac.toml")
+        .setArg(Arg{ "args" }
+                    .setDesc("Dependencies to add")
+                    .setRequired(true)
+                    .setVariadic(true))
+        .addOpt(Opt{ "--sys" }.setDesc("Use system dependency"))
+        .addOpt(Opt{ "--version" }.setDesc(
+            "Dependency version (Only used with system-dependencies)"
+        ))
+        .addOpt(Opt{ "--tag" }.setDesc("Specify a git tag").setPlaceholder("<TAG>"))
+        .addOpt(
+            Opt{ "--rev" }.setDesc("Specify a git revision").setPlaceholder("<REVISION>")
+        )
+        .addOpt(
+            Opt{ "--branch" }.setDesc("Specify a branch of the git repository").setPlaceholder("<BRANCH_NAME>")
+        )
+        .setMainFn(addMain);
+
+std::optional<std::string>
+getNextArg(
+    std::span<const StringRef>::iterator& itr,
+    const std::span<const StringRef>::iterator& end
+) {
+  if (++itr == end) {
+    return std::nullopt;
+  }
+  return std::string(*itr);
+}
+
+void
+handleDependency(std::vector<std::string>& newDeps, const std::string& dep) {
+  if (std::find(newDeps.begin(), newDeps.end(), dep) != newDeps.end()) {
+    logger::warn("The dependency `" + dep + "` is already in the poac.toml");
+  } else {
+    newDeps.push_back(dep);
+  }
+}
+
+static int
+addMain(std::span<const StringRef> args) {
+  if (args.empty()) {
+    logger::error("No dependencies to add");
+    return 0;
+  }
+
+  std::vector<std::string> newDeps = {};
+
+  bool isSystemDependency = false;
+  std::string version; // Only used with system-dependencies
+
+  std::string tag;
+  std::string rev;
+  std::string branch;
```

Comment:
I think these can be StringRef to avoid heap allocation, as we know command line arguments are valid during the entire addMain.

Reply:
I'm not so familiar with string_view. Using StringRef outputs garbage value in the toml.

Reply:
I will work on it.  Thanks for your research!